torch/csrc/jit/codegen/cuda/python_frontend/fusion_definition.h

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/kernel_cache.h>
+#include <torch/csrc/jit/codegen/cuda/python_frontend/fusion_owner.h>
+
+using NvfDataType = torch::jit::fuser::cuda::DataType;
+using NvfTensorView = torch::jit::fuser::cuda::TensorView;
+using NvfVal = torch::jit::fuser::cuda::Val;
+
+namespace nvfuser {
+
+struct RecordFunctor;
+
+enum class StateType {
+  Tensor,
+  Scalar,
+  None,
+};
+
+struct State {
+  State(StateType _stype, size_t _index) : stype(_stype), index(_index) {}
+
+  StateType stype;
+  size_t index;
```

Comment:
A bit comments describing what is `State` and what is `index` would be very helpful here.

Reply:
I added comments everywhere.  Let me know if it still doesn't seem clear.